fish-rust/src/wutil/mod.rs

```diff
@@ -81,6 +81,10 @@ pub fn perror(s: &str) {
     let _ = stderr.write_all(b"\n");
 }
 
+pub fn perror_io(s: &str, e: &io::Error) {
```

Comment:
We can't overload `perror` to take either `str` or `LibcError` but we can define a `Perror` trait and implement it for either type (`"fcntl".perror()` looks funny alright).

Reply:
If this is not used over FFI you can just have both be `impl Debug`, no need to reinvent [`std::error::Error`](https://doc.rust-lang.org/std/error/trait.Error.html)

Reply:
Debug wouldn't work because the implementation for `str` would do what `perror()` does today.
Of course eventually we want to get rid of `perror("fcntl")` etc. entirely but I wonder what's the best way to get there. My gut feeling is to do it after the port